File: Project Euler #005 - Smallest multiple/SmallestMultiple.java
import java.util.*;

public class SmallestMultiple {

    public static long gcd (long a, long b) {
        while (b > 0) {
            long x = a % b;
            a = b;
            b = x;
        }
        return a;
    }
    
    public static long getSmallestMultiple (long n) {
        long res = 1;
        for (long i = 2; i <= n; i++) {
            res = (i*(res))/gcd(i,res);
        }
        return res;
    }
    
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            long n = in.nextLong();
            System.out.println(getSmallestMultiple(n));
        }
    }
}

File: Project Euler #005 - Smallest multiple/smallest_multiple.cpp
#include <iostream>

using namespace std;

int gcd(int a, int b) {
    while (b > 0) {
        int x = a % b;
        a = b;
        b = x;
    }
    return a;
}

int smallest_multiple(int n)
{
    int res = 1;
    for (int i = 2; i <= n; i++) {
        res = (i*(res))/gcd(i,res);
    }
    return res;
}

int main(){
    int t;
    cin >> t;
    for(int a0 = 0; a0 < t; a0++){
        int n;
        cin >> n;
        cout << smallest_multiple(n) << endl;
    }
    return 0;
}
